packages/core-contracts/contracts/mocks/proxy/UniversalProxyImplementationMockA.sol

```diff
@@ -0,0 +1,40 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "../../proxy/UniversalProxyImplementation.sol";
+
+contract UniversalProxyImplementationMockA is UniversalProxyImplementation {
```

Comment:
Do you think that both `UniversalProxyImplementationMockA` and `UniversalProxyImplementationMockB` could inherit from a `UniversalProxyImplementationMock` that sets the implementation in slot 0 and implements the implementation setter/getter? That way, mock A and B don't need to do this weird slot shifting; they just don't use the first slot.

Reply:
in that case I prefer to keep it visible that there's a need to set "manually" the implementation location